rosbag2_cpp/include/rosbag2_cpp/cache/cache_buffer_interface.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_CPP__CACHE__CACHE_BUFFER_INTERFACE_HPP_
+#define ROSBAG2_CPP__CACHE__CACHE_BUFFER_INTERFACE_HPP_
+
+#include <memory>
+#include <vector>
+
+#include "rosbag2_cpp/visibility_control.hpp"
+#include "rosbag2_storage/serialized_bag_message.hpp"
+
+namespace rosbag2_cpp
+{
+namespace cache
+{
+using buffer_element_t = std::shared_ptr<const rosbag2_storage::SerializedBagMessage>;
+class ROSBAG2_CPP_PUBLIC CacheBufferInterface
+{
+public:
+  virtual ~CacheBufferInterface() {}
```

Comment:
Please add docstrings for class-level and methods of both `CacheBufferInterface` and `MessageCacheInterface`. These should clearly explain the expected behavior of these interfaces on the opaque API level. Implementing classes can add more documentation if it differs in some way from the virtual interface.